external/CMakeLists.txt

```
@@ -0,0 +1,115 @@
+cmake_minimum_required(VERSION 3.24)
+include(ExternalProject)
+
+project(schrodinger_sketcher_external)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+if(MSVC)
+  # Workaround until boost 1.83 which fixes the issue with std::unary_function
+  add_definitions(-D_HAS_AUTO_PTR_ETC=0)
+endif()
+
+# Collect dependency versions
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/versions.json VERSIONS)
+string(JSON BOOST_VERSION GET ${VERSIONS} boost)
+string(JSON FMT_VERSION GET ${VERSIONS} fmt)
+string(JSON QT_VERSION GET ${VERSIONS} qt)
+string(JSON RDKIT_VERSION GET ${VERSIONS} rdkit)
+string(JSON ZSTD_VERSION GET ${VERSIONS} zstd)
+
+ExternalProject_Add(
+  fmt
+  GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+  GIT_TAG ${FMT_VERSION}
+  GIT_SHALLOW TRUE
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/fmt-${FMT_VERSION}
+    -DCMAKE_BUILD_TYPE=Release -DFMT_DOC=OFF -DFMT_TEST=OFF -DFMT_INSTALL=ON)
+
+ExternalProject_Add(
+  zstd
+  GIT_REPOSITORY https://github.com/facebook/zstd.git
+  GIT_TAG v${ZSTD_VERSION}
+  GIT_SHALLOW TRUE
+  SOURCE_SUBDIR build/cmake
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/zstd-${ZSTD_VERSION}
+    -DCMAKE_BUILD_TYPE=Release -DZSTD_BUILD_SHARED=OFF
+    -DZSTD_BUILD_PROGRAMS=OFF)
+
+set(BOOST_INCLUDE_LIBS
+    # RDKit required libraries
+    assign
+    crc
+    format
+    flyweight
+    graph
+    iostreams
+    multi_array
+    multiprecision
+    program_options
+    serialization
+    # rdkit_extensions required libraries
+    beast
+    bimap
+    filesystem
+    test)
+# Use generator expression to sidestep CMAKE_ARGS removing semicolons
+string(REPLACE ";" "$<SEMICOLON>" BOOST_INCLUDE_LIBS "${BOOST_INCLUDE_LIBS}")
+ExternalProject_Add(
+  boost
+  GIT_REPOSITORY https://github.com/boostorg/boost.git
+  GIT_TAG boost-${BOOST_VERSION}
+  GIT_SHALLOW TRUE
+  CMAKE_ARGS
+    -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/boost-${BOOST_VERSION}
+    -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF
```

To allow boost to be found on windows (doesn't affect Linux or Mac, since this is already the default for them):
```suggestion
    -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DBOOST_INSTALL_LAYOUT=system
```

---

Offlined with Ric; I don't need INSTALL_LAYOUT to get this to work on the windows runner. We may need to deal with turning threading off for the WASM build, but kicked that conversation out to the WASM case.